website/blog/2025/06-25-enhancing-meltdown-protection-with-dependency-watchdog-annotations.md

```diff
@@ -0,0 +1,42 @@
+---
+title: "Enhancing Meltdown Protection with Dependency-Watchdog Annotations"
+linkTitle: "Enhancing Meltdown Protection with Dependency-Watchdog Annotations"
+newsSubtitle: June 25, 2025
+publishdate: 2025-06-25
+authors:
+- avatar: https://avatars.githubusercontent.com/ashwani2k
+  login: ashwani2k
+  name: Ashwani Kumar
+aliases: ["/blog/2025/06/25/enhancing-meltdown-protection-with-dependency-watchdog-annotations"]
+---
+
+Gardener's `dependency-watchdog` is a crucial component for ensuring cluster stability. During infrastructure-level outages where worker nodes cannot communicate with the control plane, it activates a "meltdown protection" mechanism. This involves scaling down key control plane components like the `machine-controller-manager` (MCM), `cluster-autoscaler` (CA), and `kube-controller-manager` (KCM) to prevent them from taking incorrect actions based on stale information, such as deleting healthy nodes that are only temporarily unreachable.
+
+### The Challenge: Premature Scale-Up During Reconciliation
+
+Previously, a potential race condition could undermine this protection. While `dependency-watchdog` scaled down the necessary components, a concurrent `Shoot` reconciliation, whether triggered manually by an operator or by other events, could misinterpret the situation. The reconciliation logic, unaware that the scale-down was a deliberate protective measure, would attempt to restore the "desired" state by scaling the `machine-controller-manager` and `cluster-autoscaler` back up.
```

Comment:
Will shoot reconciliation only being back MCM and CA, and not KCM? This is what I've understood from here

Reply:
Currently kCM is already handled via static value which are also replaced, but the behavior is visible only for MCM/CA.